libpod/oci.go

```diff
@@ -349,27 +347,8 @@ func (r *OCIRuntime) createOCIContainer(ctr *Container, cgroupParent string) (er
 	childStartPipe.Close()
 
 	// Move conmon to specified cgroup
-	if os.Getuid() == 0 {
-		if r.cgroupManager == SystemdCgroupsManager {
-			unitName := createUnitName("libpod-conmon", ctr.ID())
-
-			logrus.Infof("Running conmon under slice %s and unitName %s", cgroupParent, unitName)
-			if err = utils.RunUnderSystemdScope(cmd.Process.Pid, cgroupParent, unitName); err != nil {
-				logrus.Warnf("Failed to add conmon to systemd sandbox cgroup: %v", err)
-			}
-		} else {
-			cgroupPath := filepath.Join(ctr.config.CgroupParent, fmt.Sprintf("libpod-%s", ctr.ID()), "conmon")
-			control, err := cgroups.New(cgroups.V1, cgroups.StaticPath(cgroupPath), &spec.LinuxResources{})
-			if err != nil {
-				logrus.Warnf("Failed to add conmon to cgroupfs sandbox cgroup: %v", err)
-			} else {
-				// we need to remove this defer and delete the cgroup once conmon exits
-				// maybe need a conmon monitor?
-				if err := control.Add(cgroups.Process{Pid: cmd.Process.Pid}); err != nil {
-					logrus.Warnf("Failed to add conmon to cgroupfs sandbox cgroup: %v", err)
-				}
-			}
-		}
+	if err := r.moveConmonToCgroup(ctr, cgroupParent, cmd); err != nil {
```

Comment:
Don't bother refactoring `oci.go` - just make the entire file linux-only. This is not going to be shared code.

Reply:
> Don't bother refactoring `oci.go` - just make the entire file linux-only. This is not going to be shared code.

Really?  I'd expect we'll want the same architecture and the same runtime calls on all OSes.  What needs to change besides the way `conmon` notices the container-process exit and how we port terminal file descriptors to Windows?

Reply:
From earlier discussions, it sounded like Windows had its own low-level shim for this code, which we'd likely be using instead... I also don't know if we're willing to port Conmon to windows (though we could always make a `windowsmon` that acts exactly identical in command line argument handling but calls the appropriate Windows functions.

Reply:
> From earlier discussions, it sounded like Windows had its own low-level shim for this code, which we'd likely be using instead...

Do we know how different it is from the rest of what we have here?  The API for this file:

```console
$ grep '^[A-Za-z]' libpod/oci.go 
package libpod
import (
const (
type OCIRuntime struct {
type syncInfo struct {
func newOCIRuntime(name string, path string, conmonPath string, conmonEnv []string, cgroupManager string, tmpDir string, logSizeMax int64, noPivotRoot bool) (*OCIRuntime, error) {
func newPipe() (parent *os.File, child *os.File, err error) {
func createUnitName(prefix string, name string) string {
func waitContainerStop(ctr *Container, timeout time.Duration) error {
func waitPidsStop(pids []int, timeout time.Duration) error {
func (r *OCIRuntime) createContainer(ctr *Container, cgroupParent string) (err error) {
func (r *OCIRuntime) createOCIContainer(ctr *Container, cgroupParent string) (err error) {
func (r *OCIRuntime) updateContainerStatus(ctr *Container) error {
func (r *OCIRuntime) startContainer(ctr *Container) error {
func (r *OCIRuntime) killContainer(ctr *Container, signal uint) error {
func (r *OCIRuntime) stopContainer(ctr *Container, timeout uint) error {
func (r *OCIRuntime) deleteContainer(ctr *Container) error {
func (r *OCIRuntime) pauseContainer(ctr *Container) error {
func (r *OCIRuntime) unpauseContainer(ctr *Container) error {
func (r *OCIRuntime) execContainer(c *Container, cmd, capAdd, env []string, tty bool, user, sessionID string) (*exec.Cmd, error) {
func (r *OCIRuntime) execStopContainer(ctr *Container, timeout uint) error {
```

Sounds fairly OS-agnostic.  Obviously if someone else ends up maintaining the Windows code it won't matter to us.  But if the Windows shim ends up living in this repo it seems like it would be much easier to try and consolidate to have the fewest differences possible. 

> ... though we could always make a `windowsmon` that acts exactly identical in command line argument handling but calls the appropriate Windows functions...

This sounds best to me.

Reply:
There is a fair bit of Linux-specific handling around creation (user namespace handling for containers/storage). I don't think that's going to have any Windows parallel. A lot of the rest might well be common, though.

Reply:
im hearing .... leave it at is.

Reply:
For now, that seems safe.